docs/codeactions/code-actions.md

```diff
@@ -0,0 +1,96 @@
+---
+id: codeactions
+title: Code Actions
+---
+
+Whenever you are stuck or unsure, please open an issue or [ask us on
+Discord](https://discord.gg/DwTc8xbNDd) or [on our Matrix
+bridge](https://matrix.to/#/#scalameta:metals-contributors). This project
+follows [Scalameta's contribution
+guidelines](https://github.com/scalameta/scalameta/blob/master/CONTRIBUTING.md)
+and the [Scala CoC](https://scala-lang.org/conduct/).
+
+Code Actions are there to provide refactoring or rewriting of your current code in order to enhance its readability or save you some time.
+
+The common way of using them, is to click on the yellow light bulb appearing on your code, near the cursor and then, to pick the desired action to peform on the code.
+
+Here is a list of the code actions supported by Metals. Some of these code actions support much more complex use cases, which you would eventually discover through your code, by clicking on the magic 💡, as it appears, to get your surprise!
+
+## Convert To Named Arguments
+
+This code action adds the name to parameters passed to a method call or a class instant.
+
+[Convert To Named Arguments Code Action](https://github.com/scalameta/metals/docs/codeactions/gifs/ConvertToNamedArguments.gif)
+
+
+## Create Companion Object
+
+If you have a `trait` or a `class` definition, you can invoke this code action for creating its companion object.
+
+[Create Companion Object Code Action](https://github.com/scalameta/metals/docs/codeactions/gifs/CreateCompanionObject.gif)
+
+## Create New Symbol
+
+This is for creating a definition for a symbol which is used in your code, but not defined anywhere.
+
+[Create New Symbol Code Action](https://github.com/scalameta/metals/docs/codeactions/gifs/CreateNewSymbol.gif)
+
+## Extract Member
+
+Extract a class, trait, or object definition to a new file.
+
+[Extract Member Code Action](https://github.com/scalameta/metals/docs/codeactions/gifs/ExtractMember.gif)
+
+## Rename File
+
+Renames a file to match that of the prominent type defined in it.
+
+[Rename File](https://github.com/scalameta/metals/docs/codeactions/gifs/RenameFile.gif)
+
+### Extract Value
+
+If you think your code would look nicer, if the calcuation of a value is extracted and assined to a new `val`; this code action would come to your aid.
+[Extract Value](https://github.com/scalameta/metals/docs/codeactions/gifs/ExtractValue.gif)
+
+### map, flatMap, withFilter to For Yield
+
+It converts a chain of `map`, `flatMap`, `filter` and `filterNot` methods into a for comprehension
+
+[To For Comprehension](https://github.com/scalameta/metals/docs/codeactions/gifs/FlatMapToForComprehension.gif)
+
+## Implement Abstract Members of the Parent Type
+
+Upon inherting from a type, you also have to implement its abstract members. But manuallylooking them all up and copying their signature is time consuming, isn't it? You can just use this code action instead.
```

Comment:
```suggestion
Upon inherting from a type, you also have to implement its abstract members. But manually looking them all up and copying their signature is time consuming, isn't it? You can just use this code action instead.
```

Reply:
The informal descriptions are nowadays very popular in technical manuals and blog posts. They entertain and engage the user as they go through the article. So they increase the chance of the article for getting read to the end.

Reply:
But people are different. However you write it, there will still be people who won't be ok with it, won't there?